src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Rsa.cs

```diff
@@ -9,6 +12,117 @@ internal static partial class Interop
     internal static partial class Crypto
     {
         [DllImport(Libraries.CryptoNative)]
+        internal static extern SafeRsaHandle RsaCreate();
+
+        [DllImport(Libraries.CryptoNative)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool RsaUpRef(IntPtr rsa);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void RsaDestroy(IntPtr rsa);
+
+        [DllImport(Libraries.CryptoNative)]
         internal static extern SafeRsaHandle DecodeRsaPublicKey(byte[] buf, int len);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal extern static int RsaPublicEncrypt(
+            int flen,
+            byte[] from,
+            byte[] to,
+            SafeRsaHandle rsa,
+            [MarshalAs(UnmanagedType.Bool)] bool useOaepPadding);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal extern static int RsaPrivateDecrypt(
+            int flen,
+            byte[] from,
+            byte[] to,
+            SafeRsaHandle rsa,
+            [MarshalAs(UnmanagedType.Bool)] bool useOaepPadding);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int RsaSize(SafeRsaHandle rsa);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int RsaGenerateKeyEx(SafeRsaHandle rsa, int bits, SafeBignumHandle e);
+
+        [DllImport(Libraries.CryptoNative)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool RsaSign(int type, byte[] m, int m_len, byte[] sigret, out int siglen, SafeRsaHandle rsa);
+
+        [DllImport(Libraries.CryptoNative)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool RsaVerify(int type, byte[] m, int m_len, byte[] sigbuf, int siglen, SafeRsaHandle rsa);
+
+        internal static RSAParameters ExportRsaParameters(SafeRsaHandle key, bool includePrivateParameters)
+        {
+            Debug.Assert(
+                key != null && !key.IsInvalid,
+                "Callers should check the key is invalid and throw an exception with a message");
+
+            if (key == null || key.IsInvalid)
```

Comment:
I think I was trying to avoid having an SR reference in the common code.  So the caller would hopefully throw the exception with a meaningful message, the assert is to report that the calling code was naughty, and the runtime if was to prevent bad things from happening.